### PR TITLE
Detail page design

### LINKED ISSUE #11: Fix Filter By Functionality
Filter by menu is showing in a wrong place. Kindly Fix it.

image

### PR DESCRIPTION
**Pull request by zara19013**
this is design page, in which api integration is done, and fetching countries details

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -85,41 +33,33 @@ function App() {
...
-    </div>
-  </div>
-  </div>
-
-  </RootContext.Provider>
+      <div className={`App ${colorMode === "dark" ? "dark" : ""}`}>
+        {/* <div className="App"> */}
+          <Header />
+
+          {/* <Filter countriesData={countriesData} onFilterChange={handleFilterChange} /> */}
```
Comment (asad-shayansolution) [inline src/App.js]: remove this commented code

File: src/components/Detail/Detail.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
+// layout dessign
+// import React, { useEffect, useState } from 'react';
+// import './detail.css';
+// import App from '../../App';
+
+// function DetailsPage() {
+//   const countryData = {
+//     name: 'Algeria',
```
Comment (asad-shayansolution) [inline src/components/Detail/Detail.js]: remove this commented code.

File: src/components/Detail/Detail.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  const queryParams = new URLSearchParams(location.search);
+  const countryCode = queryParams.get("code");
+
+  const [countryData, setCountryData] = useState({});
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    axios
+      .get(`https://restcountries.com/v3.1/alpha/${countryCode}`)
+      // .get(`https://restcountries.com/v3.1/alpha/pk`)
```
Comment (asad-shayansolution) [inline src/components/Detail/Detail.js]: the PR's code should not contain any unnecessary commented code.

File: src/components/homepage/Card.css [hunk 4]
```diff
@@ -1,64 +1,61 @@
...
-  padding-bottom: 20px; 
-
+  justify-content: space-between;
+  padding-bottom: 20px;
 
-   .flag {
+  .flag {
     width: 270px;
-    height: 310 auto ;
+    height: 310 auto;
```
Comment (asad-shayansolution) [inline src/components/homepage/Card.css]: 310 what? pixels?

File: src/darkmode.css [hunk 5]
```diff
@@ -34,74 +34,80 @@
...
 }
 
-
-
-    .App.dark .flag
-{
+.App.dark .flag {
   width: 270px;
-  height: 310 auto ;
+  height: 310 auto;
```
Comment (asad-shayansolution) [inline src/darkmode.css]: add px after 310.

**Review by asad-shayansolution — COMMENTED** [review]
Please resovle all issue I commented on.

File: src/components/Detail/Detail.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+                </p>
+              </div>
+              <div className="rightInsideDetail">
+                <p>
+                  <strong>Top Level Domain:</strong>{" "}
+                  {countryData.tld.join(", ")}
+                </p>
+                <p>
+                  <strong>Currencies:</strong>{" "}
+                  {Object.entries(countryData.currencies)
```
Comment (imran-munawar) [inline src/components/Detail/Detail.js:82]: Object.values method to directly access the currency objects' properties Object.values(countryData.currencies)

File: src/components/Detail/Detail.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+                  <strong>Languages: </strong>
+                  {Object.values(countryData.languages).join(", ")}
+                </p>
+              </div>
+            </div>
+            <div className="borderCountries">
+              <div className="title">Border Countries:</div>
+              <div className="wrapBorderCountries">
+                {countryData.borders?.map((borderCountry, index) => (
+                  <button className="borderCountry" key={index}>
```
Comment (imran-munawar) [inline src/components/Detail/Detail.js]: please do not use index as key

**Review by imran-munawar — CHANGES_REQUESTED** [review]

File: src/components/Detail/Detail.js (hunk 6 above)
Comment (zara19013) [reply src/components/Detail/Detail.js:82]: sir in this we display the Currency in form of  i.e pakistani Rupee (Rs), and object call is invalid from this approch, like data is   USD: { name: 'US Dollar', symbol: '$' },
  EUR: { name: 'Euro', symbol: '€' },

**Review by asad-shayansolution — APPROVED** [review]

File: src/App.js [hunk 8]
```diff
@@ -1,81 +1,28 @@
...
 function App() {
   const [countriesData, setCountriesData] = useState([]);
   const [filteredCountries, setFilteredCountries] = useState([]);
   const [colorMode, setColorMode] = useState("light");
 
   useEffect(() => {
-    axios.get("https://restcountries.com/v3.1/all")
-      .then(response => {
+    axios
+      .get("https://restcountries.com/v3.1/all")
```
Comment (sameedFAST-NU) [inline src/App.js:20]: use .env file to store api adress

File: src/darkmode.css [hunk 9]
```diff
@@ -1,107 +1,72 @@
...
-    background: rgb(23, 23, 23);
-    color: white;
+  background: hsl(207, 26%, 17%);
+  color: white;
 }
 
+.App.dark .contents {
+  background: hsl(207, 26%, 17%);
+}
 .App.dark header {
```
Comment (sameedFAST-NU) [inline src/darkmode.css:10]: a better approach is to make just two classes for dark mode. one for background and one for elements(header, buttons and cards are all same color)

**Pull request merged**